.github/workflows/create-dev-branch.yaml

```
@@ -0,0 +1,142 @@
+        type: choice
+        default: "major"
+        options:
+          - "major"
+          - "minor"
+
+jobs:
+  create-branch:
+    runs-on: ubuntu-24.04
+    steps:
+      # validate branch
+      - name: Validate running branch
+        run: |
+          # we want to stop if this is not running on development branch
+          [[ "${{github.ref_name}}" =~ ^development/[0-9.]+$ ]] || exit 1
+      # create app token
+      - uses: actions/create-github-app-token@v1
+        id: app-token
+        with:
+          app-id: ${{ vars.ACTIONS_APP_ID }}
+          private-key: ${{ secrets.ACTIONS_APP_PRIVATE_KEY }}
+      # checkout
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          token: ${{ steps.app-token.outputs.token }}
+      # get version
+      - name: Get current version
+        run: |
+          source VERSION
+
+          SHORT_VERSION="$VERSION_MAJOR.$VERSION_MINOR"
+
+          echo "SHORT_VERSION=$SHORT_VERSION" >> $GITHUB_ENV
+      # calculate new version
+      - name: Calculate new version
+        run: |
+          source VERSION
+
+          if [[ "${{ inputs.bump-type }}" == "major" ]]; then
+            NEW_VERSION_MAJOR=$(($VERSION_MAJOR+1))
+            NEW_VERSION_MINOR=0
+          else
+            NEW_VERSION_MAJOR=$VERSION_MAJOR
+            NEW_VERSION_MINOR=$(($VERSION_MINOR+1))
+          fi
+
+          NEW_SHORT_VERSION="$NEW_VERSION_MAJOR.$NEW_VERSION_MINOR"
+
+          echo "NEW_SHORT_VERSION=$NEW_SHORT_VERSION" >> $GITHUB_ENV
+          echo "NEW_VERSION_MAJOR=$NEW_VERSION_MAJOR" >> $GITHUB_ENV
+          echo "NEW_VERSION_MINOR=$NEW_VERSION_MINOR" >> $GITHUB_ENV
+      # calculate new branch name
+      - name: Calculate new branch name
+        run: |
+          NEW_BRANCH_NAME="development/${{ env.NEW_SHORT_VERSION }}"
+          INIT_BRANCH_NAME="feature/init-dev-${{ env.NEW_SHORT_VERSION }}"
+
+          echo "NEW_BRANCH_NAME=$NEW_BRANCH_NAME" >> $GITHUB_ENV
+          echo "INIT_BRANCH_NAME=$INIT_BRANCH_NAME" >> $GITHUB_ENV
+      # create development branch
+      - name: Create new branch
+        run: |
+          git checkout -b "${{ env.NEW_BRANCH_NAME }}"
+      # push development branch
+      - name: Push new branch
+        run: |
+          git config --global user.email ${{ github.actor }}@scality.com
+          git config --global user.name ${{ github.actor }}
+          git push --set-upstream origin ${{ env.NEW_BRANCH_NAME }}
+      # create new branch
+      - name: Create an init branch
+        run: |
+          git checkout -b "${{ env.INIT_BRANCH_NAME }}"
+      # find mentions of new version: warn user
+      - name: Look for important messages
+        id: warn_user
+        run: |
+          link="https://github.com/${{ github.repository }}/blob/${{ github.ref_name }}/"
+          {
+            echo 'COMMENT<<EOF'
+            echo "# Mentions of ${{ env.NEW_SHORT_VERSION }} in ${{ github.ref_name }}"
```

To me it would be better to link on the new development branch (since it's on this branch we will want to update it)

```suggestion
          link="https://github.com/${{ github.repository }}/blob/${{ env.NEW_BRANCH_NAME }}/"
          {
            echo 'COMMENT<<EOF'
            echo "# Mentions of ${{ env.NEW_SHORT_VERSION }} in ${{ env.NEW_BRANCH_NAME }}"
```